utils/handles/signal.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __UTILS_HANDLES_SIGNAL_H__
+#define __UTILS_HANDLES_SIGNAL_H__
+
+/**
+ * We declare a function register_signal() to handle signals while the program
+ * executing. This function will catch some exceptional behavior with the
+ * program.
+ */
+
+#include <signal.h>
+#include <stdlib.h>
+#include "utils/logger_helper.h"
+
+/*
+ *  This enum is declared for consistent name of signal for c5 ross platfrom
+ * developing the naming order is followed by POSIX.1-1990 the standard signals
+ * table is at http://man7.org/linux/man-pages/man7/signal.7.html
+ *
+ */
+
+typedef enum UNIVERSAL_SIGNAL_NUM {
+  null,
+  signal_hang_up,
+  ctrl_c
+} universal_signal_num_t;
+
+#if !defined(_WIN32) && defined(__unix__) || defined(__unix) || \
+    (defined(__APPLE__) && defined(__MACH__))
+
+#include <unistd.h>
+#define CTRL_C_SIGNAL SIGINT
+#define SIGNAL_ERROR SIG_ERR
+
+#elif defined(_WIN32)
+
+#include <Windows.h>
+
+#define CTRL_C_SIGNAL CTRL_C_EVENT
+#define SIGNAL_SUCCESS TRUE
+#define SIGNAL_ERROR FALSE
+
+BOOL signal_handler_WIN(DWORD dwCtrlType);
+
+#else
+
+#error "No signal process library found"
```

Comment:
Ditto. Implementing this in header file can validate supported platforms in advance.

Reply:
So I should change this into logging functions?  Or I should add logging function and preserve the #error directive?

Reply:
Keeping `#error` for conditional build is reasonable since you already consider major UNIX-compatible systems and Windows-based environments.